brainglobe_template_builder/preproc/transform_utils.py

```diff
@@ -71,3 +73,60 @@ def apply_transform(
     # Preserve original data range and type
     transformed = np.clip(transformed, data.min(), data.max())
     return transformed.astype(data.dtype)
+
+
+def downsample_anisotropic_image_stack(
+    stack: da.Array, xy_downsampling: int, z_downsampling: int
+) -> np.ndarray:
```

Comment:
Should the stack shape be divisible by the downsampling factor or it doesn't matter?

Reply:
Updated docstring for clarity on this, and the answer is no, the borders get zero-padded :)